include/swift/Remote/MetadataReader.h

```diff
@@ -1382,38 +1395,42 @@ class MetadataReader {
         reinterpret_cast<const TargetExtendedExistentialTypeShape<Runtime> *>(
             cached->second.get()));
 
-    ExtendedExistentialTypeShapeFlags flags;
-    if (!Reader->readBytes(RemoteAddress(address), (uint8_t*)&flags,
-                           sizeof(flags)))
-      return nullptr;
 
-    // Read the size of the requirement signature.
-    uint64_t reqSigGenericSize = 0;
-    uint64_t genericHeaderSize = sizeof(GenericContextDescriptorHeader);
+    uint64_t descriptorSize;
     {
-      GenericContextDescriptorHeader header;
-      auto headerAddr = address + sizeof(flags);
-
-      if (!Reader->readBytes(RemoteAddress(headerAddr),
-                             (uint8_t*)&header, sizeof(header)))
+      using RelativeOffset = TargetRelativeDirectPointer<Runtime, const char,
+                                                         /*nullable*/ false>;
+      constexpr size_t buffer_size = sizeof(ExtendedExistentialTypeShapeFlags) +
+                                     sizeof(RelativeOffset) +
+                                     sizeof(GenericContextDescriptorHeader);
```

Comment:
Please try:

```
union {
  TargetExtendedExistentialTypeShape<Runtime> shapeHeader;
};
if (!Reader->readBytes(RemoteAddress(address), reinterpret_cast<uint8_t*>(&shapeHeader), sizeof(shapeHeader))
  return nullptr;
```

and then you should be able to just use `shapeHeader.Flags` and `shapeHeader.ReqSigHeader` without any other fuss.

Reply:
That didn't work directly because TargetExtendedExistentialTypeShape has no default constructor, but I updated the patch to do achieve something pretty close.